crates/turborepo-paths/src/absolute_system_path.rs

```diff
@@ -107,21 +108,30 @@ impl AbsoluteSystemPath {
         &self.0
     }
 
-    pub fn join_relative(&self, path: &RelativeSystemPathBuf) -> AbsoluteSystemPathBuf {
-        let path = self.0.join(path.as_path());
-        AbsoluteSystemPathBuf(path)
+    // intended for joining literals or obviously single-token strings
+    pub fn join_literal(&self, segment: &str) -> AbsoluteSystemPathBuf {
```

Comment:
I have a slight issue with the naming here since we allow non-literals and we make use of this behavior. I feel like `join_path_component` is slightly more honest.

Reply:
went with `join_component`